openedx/core/djangoapps/course_apps/rest_api/v1/views.py

```diff
@@ -133,11 +135,13 @@ def get(self, request: Request, course_id: str):
         """
         course_key = CourseKey.from_string(course_id)
         course_apps = CourseAppsPluginManager.get_apps_available_for_course(course_key)
+        course_apps_status = CourseAppStatus.get_all_app_status_data_for_course(course_key)
```

Comment:
Here we are expecting all apps to be available for _happy_ scenario right? and if that fails, we are calling `is_course_app_enabled` < which actually again tries to pull data from `CourseAppStatus` and then fall back. 

Do you think this code needs a little cleanup too?